src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return (
+    sourceString
+      .split(';')
+      .map((item) => item.trim())
+      .filter((item) => item)
+      .reduce((acc, item) => {
+        const [property, value] = item.split(':').map((items) => items.trim());
+
+        acc[property] = value;
+
+        return acc;
+      }, {}) || {}
```

Comment:
The use of `|| {}` at the end of the reduce function is unnecessary. The reduce function already initializes the accumulator as an empty object (`{}`), so the logical OR is redundant. It could potentially mask errors by returning an empty object instead of allowing an error to propagate if the reduce function fails.